pandas/io/excel.py

```diff
@@ -66,27 +69,162 @@ def get_writer(engine_name):
         raise ValueError("No Excel writer '%s'" % engine_name)
 
 
+class BaseFile(object):
+    """Base class for excel readers
+
+    A file class can be initialized even if the engine is not installed.
+    If the engine is not installed, io_class and _open_workbook are
+    type(None) and None respectively. When attempting to use open_workbook
+    while _open_workbook is None, the relevant ImportError is raised. When
+    calling is_type or is_ext, False is retunred when the reader is not
+    installed.
+    """
+    # added in order to deal with both Open Document Format and MS Excel
+    # fil types, see PR #9070
+
+    def __init__(self, engine, extensions):
```

Comment:
make a class-method ``BaseFile.create`` which returns the appropriate sub-class based on the extension. Then you don't really need much of this class/logic.